rholang/src/main/scala/coop/rchain/rholang/interpreter/matcher/StreamT.scala

```diff
@@ -175,3 +190,17 @@ trait StreamTInstances2 {
       def zero[A](state: Stream[A]): Boolean = state.isEmpty
     }
 }
+
+private trait StreamTMonadErrorMonad[F[_]]
+    extends MonadError[StreamT[F, ?], Unit]
```

Comment:
The usefulness of this is limited if you can't define the Error type - it's hardcoded to `Unit` right?

Reply:
This is what allows deriving `_short[NonDetFreeMapWithCost]`, just like we had it for OptionalFreeMapWithCost. Notice there's 2x MonadError in the stack(s)

Reply:
What I meant was that if you add type parameter for error you get `MonadError[StreamT[F, ?], E]` and you should be able to still get `_short[NonDetFreeMapWithCost]`

Reply:
The MonadError[F, Unit] instance targets the 'StreamT/OptionT' level of the monad stack and is a **different effect** than the one arising from the `Either` in the stack (giving rise to `_error`).

Reply:
Sure, but you can get `MonadError[F, Unit]` from `MonadError[F, E]`.

Reply:
Again, it's not about the type - it's about the stack level.
`_short[F].raiseError` and `_error[F].raiseError` fail the computation *differently* - the latter is just `Left(OutOfPholgistonError)`, the former is  `Right((Cost, None))` or `Right((Cost, Stream.empty))`.

To achieve that alst bit, I needed to copy the derivation rule for OptionT from cats and `s/OptionT/StreamT`.